test/torchaudio_unittest/prototype/functional/functional_test_impl.py

```diff
@@ -545,3 +546,288 @@ def test_simulate_rir_ism_multi_band(self, channel):
             expected[i, 0 : room.rir[i][0].shape[0]] = torch.from_numpy(room.rir[i][0])
         actual = F.simulate_rir_ism(room_dim, source, mic_array, max_order, absorption)
         self.assertEqual(expected, actual, atol=1e-3, rtol=1e-3)
+
+    @parameterized.expand(
+        [
+            (0.1, 0.2, (2, 1, 2500)),  # both float
+            # Per-wall
+            (torch.rand(6), 0.2, (2, 1, 2500)),
+            (0.1, torch.rand(6), (2, 1, 2500)),
+            (torch.rand(6), torch.rand(6), (2, 1, 2500)),
+            # Per-band and per-wall
+            (torch.rand(4, 6), 0.2, (2, 4, 2500)),
+            (0.1, torch.rand(4, 6), (2, 4, 2500)),
+            (torch.rand(4, 6), torch.rand(4, 6), (2, 4, 2500)),
+        ]
+    )
+    def test_ray_tracing_output_shape(self, absorption, scattering, expected_shape):
+        room_dim = torch.tensor([20, 25, 30], dtype=self.dtype)
+        mic_array = torch.tensor([[2, 2, 2], [8, 8, 8]], dtype=self.dtype)
+        source = torch.tensor([7, 6, 5], dtype=self.dtype)
+        if isinstance(absorption, torch.Tensor):
+            absorption = absorption.to(self.dtype)
+        if isinstance(scattering, torch.Tensor):
+            scattering = scattering.to(self.dtype)
+
+        num_rays = 100
+
+        hist = F.ray_tracing(
+            room=room_dim,
+            source=source,
+            mic_array=mic_array,
+            num_rays=num_rays,
+            absorption=absorption,
+            scattering=scattering,
+        )
+
+        assert hist.shape == expected_shape
+
+    def test_ray_tracing_input_errors(self):
```

Comment:
need more of these. currently the implementation does not reject the invalid shapes like empty tensors.